src/modules/trade/hooks/useTradeRouteContext.ts

```diff
@@ -0,0 +1,18 @@
+import { useMemo } from 'react'
+
+import { useTradeState } from './useTradeState'
+
+import { TradeUrlParams } from '../types/TradeRawState'
+
+export function useTradeRouteContext(): TradeUrlParams {
+  const { state } = useTradeState()
+
+  return useMemo(
+    () => ({
+      inputCurrencyId: state?.inputCurrencyId || undefined,
+      outputCurrencyId: state?.outputCurrencyId || undefined,
+      chainId: state?.chainId?.toString(),
+    }),
+    [state]
```

Comment:
Just to avoid excessive re-renders:

```suggestion
  const { state } = useTradeState()
  const { inputCurrencyId, outputCurrencyId, chainId } = state || {}

  return useMemo(
    () => ({
      inputCurrencyId: inputCurrencyId || undefined,
      outputCurrencyId: outputCurrencyId || undefined,
      chainId: chainId?.toString(),
    }),
    [inputCurrencyId, outputCurrencyId, chainId]
```